### ISSUE TITLE
How to pass data between different templates in the same layout

### ISSUE THREAD

**Issue by sf-wind** [meteorhacks/flow-layout#30]
I have a page with two sections, header and main, each has its own template. Some fairly expensive reactive data is needed by both templates. If the visitor browse another page, the reactive data should be destroyed. Is it possible to attach that reactive data to the route/layout? 

Without FlowLayout's support, it is possible to create a parent template that includes both header and main templates, and pass the data to them. That parent template is actually the layout template. It means we need to create a separate layout template when some data is different. Does that make sense?

Say, I have the following layout:
FlowLayout.render ('layout', {header : 'headerTemplate', main : 'mainTemplate'});

I need to pass some complicated object data between headerTemplate and mainTemplate. Does it mean I need to create a new "layout" template?

**Comment by arunoda** [comment]
Don't pass data via FlowLayout. Get data from the template itself. For an example, if your data inside the collection, simply access the collection in both of your templates.

**Comment by sf-wind** [comment]
The data is from an HTTP call to an external server. Because client side HTTP call has to be asynchronous, the data needs to be saved to a reactive variable and then the template re-renders itself. 

I'd like to only make one HTTP call and render both templates. 

Any way to achieve this?

**Comment by mcissel** [comment]
Save the HTTP request results in a Session variable. Use Session.get('results') in a helper in each template.

**Comment by sf-wind** [comment]
Yes, this is one method. I just want to avoid using global session variable as much as possible. It would be good that the router can provide such a functionality.

**Comment by frozeman** [comment]
+1

This for example can be necessary to set a templates state from the route. So passing data should definitely be possible, even if it is not reactive.
